crates/swc_css_lints/src/rules/custom_property_no_missing_var_function.rs

```diff
@@ -0,0 +1,57 @@
+use swc_css_ast::*;
+use swc_css_visit::{Visit, VisitWith};
+
+use crate::rule::{visitor_rule, LintRule, LintRuleContext};
+
+pub fn custom_property_no_missing_var_function(ctx: LintRuleContext<()>) -> Box<dyn LintRule> {
+    visitor_rule(
+        ctx.reaction(),
+        CustomPropertyNoMissingVarFunction {
+            ctx,
+            ..Default::default()
+        },
+    )
+}
+
+#[derive(Debug, Default)]
+struct CustomPropertyNoMissingVarFunction {
+    ctx: LintRuleContext<()>,
+
+    in_declaration: bool,
+    in_declaration_value: bool,
+    in_var_function: bool,
+}
+
+impl Visit for CustomPropertyNoMissingVarFunction {
+    fn visit_declaration(&mut self, declaration: &Declaration) {
+        self.in_declaration = true;
```

Comment:
I think we should preserve old state, as declaration can be nested.

Reply:
https://docs.rs/swc_css_ast/latest/swc_css_ast/enum.ComponentValue.html#variant.DeclarationOrAtRule
https://docs.rs/swc_css_ast/latest/swc_css_ast/enum.DeclarationOrAtRule.html
https://docs.rs/swc_css_ast/latest/swc_css_ast/struct.Declaration.html

Reply:
`ComponentValue::DeclarationOrAtRule(DeclarationOrAtRule::Declaration(Declaration {}))`